daemon/container.go

```diff
@@ -326,6 +326,14 @@ func (container *Container) Start() (err error) {
 		return nil
 	}
 
+	if container.RemovalInProgress {
+		return fmt.Errorf("Container is already being removed. Can not start it.")
+	}
+
+	if container.Dead {
+		return fmt.Errorf("Container is already dead. Can not start it.")
```

Comment:
Dead means that container can't be started now. One can only try to remove it again. (After previous removal attempt had failed). A dead container might have partially cleaned up its resources already and only some of the resources still might be there.

For example, If I bind mount a container device somewhere, then container removal will fail. It will be in "Removal In Progress" state for 10 seconds and then transition into "Dead" state. 

A user can try to figure out what was wrong, why container removal failed. Fix that condition and try removal again. 

Had to keep a separate state to mark that container is partially destroyed otherwise one might want to start it.